arm-servicefabric/2016-09-01/swagger/servicefabric.json

```diff
@@ -0,0 +1,1041 @@
+          "additionalProperties": {
+            "$ref": "#/definitions/PaasServiceTypeHealthPolicy"
+          }
+        }
+      }
+    },
+    "PaasClusterUpgradeHealthPolicy": {
+      "description": "Paas cluster upgrade health policy",
+      "properties": {
+        "maxPercentUnhealthyNodes": {
+          "type": "integer"
+        },
+        "maxPercentUnhealthyApplications": {
+          "type": "integer"
+        },
+        "applicationHealthPolicies": {
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/PaasApplicationHealthPolicy"
+          }
+        }
+      }
+    },
+    "PaasServiceTypeDeltaHealthPolicy": {
+      "description": "Paas service type delta health policy",
+      "properties": {
+        "maxPercentDeltaUnhealthyServices": {
+          "type": "integer"
+        }
+      }
+    },
+    "PaasApplicationDeltaHealthPolicy": {
+      "description": "Paas application delta health policy",
+      "properties": {
+        "defaultServiceTypeDeltaHealthPolicy": {
+          "$ref": "#/definitions/PaasServiceTypeDeltaHealthPolicy"
+        },
+        "serivceTypeDeltaHealthPolicies": {
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/PaasServiceTypeDeltaHealthPolicy"
+          }
+        }
+      }
+    },
+    "PaasClusterUpgradeDeltaHealthPolicy": {
+      "description": "Paas cluster upgrade delta health policy",
+      "properties": {
+        "maxPercentDeltaUnhealthyNodes": {
+          "type": "integer"
+        },
+        "maxPercentUpgradeDomainDeltaUnhealthyNodes": {
+          "type": "integer"
+        },
+        "maxPercentDeltaUnhealthyApplications": {
+          "type": "integer"
+        },
+        "applicationHealthPolicies": {
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/PaasApplicationDeltaHealthPolicy"
+          }
+        }
+      }
+    },
+    "PaasClusterUpgradePolicy": {
+      "description": "Paas cluster upgrade policy",
+      "properties": {
+        "overrideUserUpgradePolicy": {
+          "type": "boolean"
+        },
+        "forceRestart": {
+          "type": "boolean"
+        },
+        "upgradeReplicaSetCheckTimeout": {
+          "type": "string"
+        },
+        "healthCheckWaitDuration": {
+          "type": "string"
+        },
+        "healthCheckStableDuration": {
+          "type": "string"
+        },
+        "healthCheckRetryTimeout": {
+          "type": "string"
+        },
+        "upgradeTimeout": {
+          "type": "string"
+        },
+        "upgradeDomainTimeout": {
+          "type": "string"
+        },
+        "healthPolicy": {
+          "$ref": "#/definitions/PaasClusterUpgradeHealthPolicy"
+        },
+        "deltaHealthPolicy": {
+          "$ref": "#/definitions/PaasClusterUpgradeDeltaHealthPolicy"
+        }
+      }
+    },
+    "DiagnosticsStorageAccountConfig": {
```

Comment:
If this is required, I am assuming, fields with in this is also required, like primaryAccessKey etc, in case properties in this class are not marked required, SDK will not validate the properties on client side and call will reach service just to be failed.